src/query/ast/src/parser/statement.rs

```diff
@@ -516,6 +516,12 @@ pub fn statement_body(i: Input) -> IResult<Statement> {
             })
         },
     );
+    let use_catalog = map(
+        rule! {
+            (USE ~ CATALOG) ~ #ident
+        },
```

Comment:
remove the parenthesis will fix the matched as `use database` issues
```rust
        rule! {
            USE ~ CATALOG ~ #ident
        },
```

Reply:
I removed the parenthesis but still the same result

Reply:
I made a mistake. It's okay to keep the parenthesis.
This problem should be caused by the MySQL client, you can use the [BendSQL](https://github.com/databendlabs/bendsql) client to test this case. Add please add some test cases to this file `src/query/ast/tests/it/parser.rs` for verification.